packages/gatsby/index.d.ts

```diff
@@ -987,7 +987,10 @@ type ReactProps<T extends Element> = React.DetailedHTMLProps<
   React.HTMLAttributes<T>,
   T
 >
-export interface RenderBodyArgs {
+export interface RenderBodyArgs<
+  PageContextType = Record<string, unknown>
+> {
+  loadPageDataSync: (pathname: string) => { result: PageContextType }
```

Comment:
The function gets you the page context for the `pathname` you give it, so I don't think we can use the generic `PageContextType` here as it'll be (potentially) different for each page. Current code assumes that `PageContextType` would be the same for each page. So let's do this instead?

```suggestion
  loadPageDataSync: (pathname: string) => { result: Record<string, unknown> }
```